decidim-proposals/app/controllers/decidim/proposals/proposal_votes_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    # Exposes the proposal vote resource so users can vote proposals.
+    class ProposalVotesController < Decidim::Proposals::ApplicationController
+      before_action :authenticate_user!
+
+      def create
+        @proposal = Proposal.find(params[:proposal_id])
+
+        set_votes_count_classes
+        set_vote_button_classes
+
+        @proposal.votes.create!(author: current_user)
```

Comment:
We may need some kind of design for this then. We are using AJAX to vote a comment so we may need some kind of error message inside the button where the vote creation fails. I'm opening an issue in design.